app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -3,32 +3,26 @@
 module Users
   class OmniauthCallbacksController < Devise::OmniauthCallbacksController
     skip_before_action :authenticate_user!
+    before_action :authorize_guild_member
 
-    before_action :non_cg_members
+    def discord
+      @user = User.from_omniauth(request.env['omniauth.auth'])
 
-    def yandex
-      user = from_omniauth(request.env['omniauth.auth'])
-      sign_in(user)
-      redirect_to after_sign_in_path_for(user)
+      if @user.persisted?
+        sign_in_and_redirect @user
+        set_flash_message(:notice, :success, kind: 'Discord') if is_navigational_format?
+      else
+        session['devise.discord_data'] = request.env['omniauth.auth'].except(:extra)
+        redirect_to new_user_registration_url
+      end
     end
 
     private
 
-    def non_cg_members
-      email = request.env['omniauth.auth'].info.email
-      return if email.include?('@cybergizer.com') || User.find_by(email: email)
-
-      redirect_to login_url, notice: 'Only Cybergizer Members can log in'
-    end
+    def authorize_guild_member
+      return if Discord::GuildMemberValidator.new(request, ENV['CYBERGIZER_SERVER_ID']).call
```

Comment:
I'd remove ENV from here and added to GuildMemberValidator, and also lets remove new :)